### PR TITLE
Add Type Illusion

### PR DESCRIPTION
**Pull request by HiTechFlufi**
Add Custom Metagame.

### CODE REVIEW

**Comment by ghost** [comment]
Oh look we finally have all the type combinations!

Besides that, it has been tested on exiled so it works.

Here is my personal 2-Cents on it. 1. change the nicks from cool names to the typings like grasssteel (all lowercase for no capitalization prefrence) and 2 change `onMega:` to something that is general on all form changes (im sure there is something!)

**Comment by HiTechFlufi** [comment]
Uh.. Sure, but it'll probably take a while. Mewth just finished spending 10 hours typing out every single nickname, and I don't really have the energy to go to every single line of code and change a nickname to a type.

File: config/formats.js [hunk 1]
```diff
@@ -467,6 +467,1363 @@ exports.Formats = [
 			this.add('raw', '<center><p><a title="Digimon Showdown Players Guide" href="https://1drv.ms/b/s!AvoD6RnUzzMvgmLcX1rqT8GTnEVK" target="_blank"><img src="https://yheeqg.bn1303.livefilestore.com/y4mcqvreFTM4wIjnVqqSI98LZXT-lFFCxPBYHyHGZtWJTIGuah-spBzqvXbLiWvyJgDnrjSRFziff59ZLLBEtm_t3ZHHeKv9AY6Ml-gQpdGMhNmzaeSt0TD8wmivpfEW81jh93LSBvdXb7-cZUH2YNJwzwiyOTsy4L2dViyJqkiOCFDZhJRIPwPEWgmD7MDJnfafmoq6SN6gEfsAXMiHh529w?width=124&amp;height=42&amp;cropmode=none" alt="" width="124" height="42" /></a></p></center>');
 		},
 	},
+	{
+		name: "[Gen 7] Type Illusion",
+		section: "SpacialGaze Metagames",
+		mod: "gen7",
+		ruleset: ['[Gen 7] OU'],
+		banlist: ["Shedinja"],
+		allowTies: true,
```
Comment (DeathlyPlays) [inline config/formats.js]: Not a function on SG as of yet

File: config/formats.js [hunk 2]
```diff
@@ -467,6 +467,1363 @@ exports.Formats = [
...
+			}
+			if (name === 'taintedmetal' && !pokemon.illusion) {
+				this.add('-start', pokemon, 'typechange', 'Dark/Steel');
+				pokemon.types = ["Dragon", "Steel"];
+			}
+			if (name === 'taintedfairy' && !pokemon.illusion) {
+				this.add('-start', pokemon, 'typechange', 'Dark/Fairy');
+				pokemon.types = ["Dragon", "Fairy"];
+			}
+			if (name === 'pixie shield' && !pokemon.illusion) {
```
Comment (DeathlyPlays) [inline config/formats.js]: Shouldn't have a space

**Review by DeathlyPlays — DISMISSED** [review]
Slight suggestions to prevent potential bugs

File: config/formats.js (hunk 1 above)
Comment (HoeenCoder) [reply config/formats.js]: `allowTies`? Wont cause any issues if its there, but yeah probably best to remove it as were still working on ties on main. We might change how its marked.

Comment (DeathlyPlays) [reply config/formats.js]: Yeah allowTies is on Exiled at the moment so he probably didn't catch that

**Pull request merged**